src/eredis.erl

```diff
@@ -118,6 +138,55 @@ q_async(Client, Command, Pid) when is_pid(Pid) ->
     Request = {request, create_multibulk(Command), Pid},
     gen_server:cast(Client, Request).
 
+%%--------------------------------------------------------------------
+%%% CREDENTIALS API
+%%--------------------------------------------------------------------
+
+-spec make_credentials(password() | credentials()) -> credentials().
+make_credentials(#{password := Password} = Credentials) ->
+    Username = maps:get(username, Credentials, undefined),
+    make_credentials(Username, Password);
+make_credentials(Password) ->
+    make_credentials(undefined, Password).
+
+-spec make_credentials(username() | undefined, password()) -> credentials().
+make_credentials(Username, Password) when
+    (Username =:= undefined orelse is_list(Username) orelse is_binary(Username)) andalso
+    (is_list(Password) orelse is_binary(Password) orelse is_function(Password, 0))
+->
+    WrappedPassword =
+        case is_function(Password, 0) of
+            true -> Password;
+            false -> eredis_secret:wrap(Password)
+        end,
```

Comment:
Tiny nit: `wrap` and `unwrap` already handle nested funs, so there should be no need for checking here.